server/src/internalClusterTest/java/org/opensearch/indices/IndicesRequestCacheIT.java

```diff
@@ -654,38 +679,547 @@ public void testCacheWithInvalidation() throws Exception {
+        }, 1, TimeUnit.SECONDS);
+    }
+
+    // staleness threshold dynamic updates should throw exceptions on invalid input
+    public void testStaleKeysCleanup_ThresholdUpdatesShouldThrowExceptionsAppropriately() throws Exception {
+        String node = internalCluster().startNode(
+            Settings.builder()
+                .put(IndicesRequestCache.INDICES_REQUEST_CACHE_STALENESS_THRESHOLD_SETTING_KEY, 0.90)
+                .put(IndicesRequestCache.INDICES_REQUEST_CACHE_CLEAN_INTERVAL_SETTING_KEY, TimeValue.timeValueMillis(1_000))
+        );
+        Client client = client(node);
+        String index1 = "index1";
+        setupIndex(client, index1);
+
+        // create first cache entry in index1
+        createCacheEntry(client, index1, "hello");
+        assertCacheState(client, index1, 0, 1);
+        assertTrue(getRequestCacheStats(client, index1).getMemorySizeInBytes() > 0);
+
+        // Update indices.requests.cache.cleanup.staleness_threshold to "10%" with illegal argument
+        try {
+            ClusterUpdateSettingsRequest updateSettingsRequest = new ClusterUpdateSettingsRequest();
+            updateSettingsRequest.persistentSettings(
+                Settings.builder().put(INDICES_REQUEST_CACHE_STALENESS_THRESHOLD_SETTING.getKey(), 10)
+            );
+            assertAcked(client().admin().cluster().updateSettings(updateSettingsRequest).actionGet());
+        } catch (Exception e) {
+            assert (e instanceof IllegalArgumentException);
+        }
+
+        // everything else should continue to work fine later on.
+        // force refresh so that it creates 1 stale key
+        flushAndRefresh(index1);
+        // sleep until cache cleaner would have cleaned up the stale key from index 2
+        assertBusy(() -> {
+            // cache cleaner should NOT have cleaned from index 1
+            assertEquals(0, getRequestCacheStats(client, index1).getMemorySizeInBytes());
+        }, 1, TimeUnit.SECONDS);
+    }
+
+    // closing the Index after caching will clean up from Indices Request Cache
+    public void testStaleKeysCleanup_ClosedIndex() throws Exception {
+        String node = internalCluster().startNode(
+            Settings.builder()
+                .put(IndicesRequestCache.INDICES_REQUEST_CACHE_STALENESS_THRESHOLD_SETTING_KEY, 0.10)
+                .put(IndicesRequestCache.INDICES_REQUEST_CACHE_CLEAN_INTERVAL_SETTING_KEY, TimeValue.timeValueMillis(3_000))
+        );
+        Client client = client(node);
+        String index = "index";
+        setupIndex(client, index);
+
+        // create first cache entry in index
+        createCacheEntry(client, index, "hello");
+        assertCacheState(client, index, 0, 1);
+        assertTrue(getRequestCacheStats(client, index).getMemorySizeInBytes() > 0);
+        assertTrue(getNodeCacheStats(client).getMemorySizeInBytes() > 0);
+
+        // force refresh so that it creates 1 stale key
+        flushAndRefresh(index);
```

Comment:
Why we need to trigger flush if the test is validating cleanup on shard close ?

Reply:
yep, good catch, this is not needed.